main.js

```diff
@@ -143,4 +148,31 @@ document.addEventListener("DOMContentLoaded", function () {
       editTodoItem(event.target.parentElement.parentElement);
     }
   });
+
+  // Save tasks to localStorage
+  function saveTasks() {
+    const tasks = [];
+    todoList.querySelectorAll("li").forEach((item) => {
+      const text = item.querySelector(".text").innerText;
+      const isChecked = item.querySelector("input[type='checkbox']").checked;
+
+      tasks.push({ text, isChecked });
+    });
```

Comment:
Instead of reading each time from DOM, try to create local state, which will have an array of items in memory.

Reading from DOM is slow operation, overall any operation with DOM, that's the reason why majority of frontend frameworks create virtual DOM and are trying to minimize operations with DOM.

Reference:
- https://habr.com/ru/companies/macloud/articles/558682/
- https://legacy.reactjs.org/docs/faq-internals.html